### PR TITLE
Roles edit 1: Added Roles update API.

### PR DESCRIPTION
**Pull request by KH-Amir-TN**
## Description

Enable users with the right set of privileges to update roles.

This PR completes **1.** 
--
~~0. Add `color:string` attribute to roles model, `null` values aren't allowed and should default to the app primary HEX color.~~
~~1. Add  `RolesController#update` to edit a new role while validating its color to be a HEX 
color.~~
> When permissions are made roles updates should change their permissions.
2. Extend the UI on Roles page in the organization settings to select a Role and show
an edit page like with room settings with a form to update name, color and turn on/off each
 role permission.
3. Extend/Add required mutations and feedbacks to integrate with the `UpdateRoleForm` 
with `RolesController#update` API.

 
### User story [Manage Roles]:
---
1. A User with the right set of privileges authenticates.
2. User goes to the Organization settings.
3. User goes to the Roles page.
4. User can select a role and:
    - Delete it (Not all roles can be deleted).
[OR]
    - Edit color, name, permissions.
[OR]
    - Add new role while providing its name and maybe color.

## Testing Steps

> 1. Pull the code.
> 2. Install the dependencies `bundle install && npm|yarn install`.
> 3. Clean the previous assets build by running `rm app/assets/builds/*` (This won't remove .keep since it's hidden).
> 4. Clean the database and tmp files for a better isolation by running `rails tmp:clear && rails db:schema:cache:clear && rails db:drop && rails db:create && rails db:migrate:with_data`
> 5. Run the linter and specs `bundle exec rubocop --parallel && bundle exec rspec && npx eslint app/javascript/* --ext .jsx,.js`
> 6. Run `./bin/dev` to run the assets builders processes and the Puma server all at once.

### CODE REVIEW

**Comment by KH-Amir-TN** [comment]
# DEPENDS ON #3574

File: app/controllers/api/v1/admin/roles_controller.rb [hunk 1]
```diff
@@ -17,6 +18,40 @@ def index
...
+
+        # POST /api/v1/roles.json
+        # Expects: { role: {:name, :color} }
+        # Returns: { data: Array[serializable objects] , errors: Array[String] }
+        # Does: Creates a role.
+
+        def create
+          role = Role.new role_params
+
+          return render_json errors: role.errors.to_a, status: :bad_request unless role.save
```
Comment (hadicheaito1) [inline app/controllers/api/v1/admin/roles_controller.rb:30]: Although this works. I prefer the simpler (easier to read) way that we have in other controllers such as in rooms_controller

File: app/controllers/api/v1/admin/roles_controller.rb [hunk 2]
```diff
@@ -17,6 +18,40 @@ def index
...
+          render_json status: :created
+        end
+
+        # POST /api/v1/:id/roles.json
+        # Expects: { role: {:name, :color} }
+        # Returns: { data: Array[serializable objects] , errors: Array[String] }
+        # Does: Updates a role.
+
+        def update
+          return render_json errors: @role.errors.to_a, status: :bad_request unless @role.update role_params
```
Comment (hadicheaito1) [inline app/controllers/api/v1/admin/roles_controller.rb:41]: Same as above comment

File: app/controllers/api/v1/admin/roles_controller.rb (hunk 1 above)
Comment (KH-Amir-TN) [reply app/controllers/api/v1/admin/roles_controller.rb:30]: Which is?

Comment (KH-Amir-TN) [reply app/controllers/api/v1/admin/roles_controller.rb:30]: Is it the  `render_error`?
If that's the case then this PR was made way before including it so I forgot to include it, will do Inshallah.

Comment (hadicheaito1) [reply app/controllers/api/v1/admin/roles_controller.rb:30]: It isn't a big deal and you don't need to change it. I am just trying to keep simple CRUD procedures following the same format is possible

**Pull request merged**